app/controllers/users_controller.rb

```diff
@@ -0,0 +1,63 @@
+class UsersController < ApplicationController
+    before_action :only_see_own_page, only: :show
+
+
+    def index
+        @users = User.all
+        @user_products = Product.where(user_id: params[:id])
+    end
+
+    def show
+    end
+
+    def create
+        auth_hash = request.env["omniauth.auth"]
+        user = User.find_by(uid: auth_hash[:uid], provider: "github")
+        if user
+            flash[:success] = "Logged in as returning user #{user.username}"
+        else
+            user = User.build_from_github(auth_hash)
+
+            if user.save
+                flash[:success] = "Logged in as new user #{user.username}"
+            else
+                head :not_found
```

Comment:
Line 24: head :note_found combined with the redirect down below on line 26 causes an error: "Render and/or redirect were called multiple times in this action. Please note that you may only call render OR redirect, and at most once per action. Also note that neither redirect nor render terminate execution of the action, so if you want to exit an action after redirecting, you need to do something like "redirect_to(...) and return". Removing this line.